flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -641,4 +610,32 @@ static void require(boolean condition, String message, Object... values) {
 		}
 	}
 
+	/**
+	 * Get dynamic properties based on two Flink configuration. If base config does not contain and target config
+	 * contains the key or the value is different, it should be added to results. Otherwise, if the base config contains
+	 * and target config does not contain the key, it will be ignored.
+	 * @param baseConfig The base configuration.
+	 * @param targetConfig The target configuration.
+	 * @return Dynamic properties as string, separated by space.
+	 */
+	static String getDynamicProperties(
+		org.apache.flink.configuration.Configuration baseConfig,
+		org.apache.flink.configuration.Configuration targetConfig) {
+
+		String[] newAddedConfigs = targetConfig.keySet().stream().flatMap(
+			(String key) -> {
+				final String baseValue = baseConfig.getString(ConfigOptions.key(key).stringType().noDefaultValue());
+				final String targetValue = targetConfig.getString(ConfigOptions.key(key).stringType().noDefaultValue());
+
+				if (!baseConfig.keySet().contains(key) || !baseValue.equals(targetValue)) {
+					return Stream.of("-" + CommandLineOptions.DYNAMIC_PROPERTY_OPTION.getOpt() + key +
+						CommandLineOptions.DYNAMIC_PROPERTY_OPTION.getValueSeparator() + targetValue);
+				} else {
+					return Stream.empty();
+				}
+			})
+			.toArray(String[]::new);
+		return org.apache.commons.lang3.StringUtils.join(newAddedConfigs, " ");
```

Comment:
I'm not sure whether `java.lang.String.join` has the same semantic of `o.a.c.lang3.StringUtils.join`, but if it is not so different, I tend to use `java.lang.String.join` :-)

Reply:
OK. I will use `java.lang.String.join` instead.